cscs-checks/microbenchmarks/mpi/osu/osu_tests.py

```diff
@@ -161,45 +93,76 @@ def set_performance_patterns(self):
         }
 
 
-class P2PBaseTest(rfm.RegressionTest):
-    exclusive_access = True
-    strict_check = False
-    num_tasks = 2
+@rfm.simple_test
+class alltoall_check(osu_latency):
+    message_size = 8
+    executable = 'osu_alltoall'
```

Comment:
The various benchmarks must be part of the library. Here should go *only* the CSCS-specific parts.

Reply:
My idea was to do not have the executables or job parameters like the number of task on the lib side. I thought the lib tests as having no meaning except that one is for latency tests and the other one for bandwidth tests. Depending of what's set in the tests that inherit from them, they could be an alltoall test over multiple nodes or a point to point test that uses two nodes.

We could put defaults on the lib side and make them point to point tests. What do you think?

Reply:
For P2P tests, `num_tasks = 2`, we know that. For collectives, we can make `num_tasks` required. The idea is that as a user I should be able to run the the benchmark from the command line without having to write a derived test.